pysat/tests/test_utils_io.py

```diff
@@ -278,23 +271,35 @@ def test_read_netcdf4_w_epoch_kwargs(self, kwargs, target):
 
         """
 
-        # TODO(#947) Expand to xarray objects once functionality is updated
-        if not self.testInst.pandas_format:
-            pytest.skip('Not yet implemented for xarray')
-
         # Create a bunch of files by year and doy
-        outfile = os.path.join(self.testInst.files.data_path,
-                               'pysat_test_ncdf.nc')
+        outfile = os.path.join(self.tempdir.name,
+                               'pysat_{:}_ncdf.nc'.format(self.testInst.name))
         self.testInst.load(date=self.stime, use_header=True)
+        if not self.testInst.pandas_format:
+            # A distinct epoch name must be used for the saved xarray dataset.
+            test_epoch = 'test_epoch'
+            kwargs['epoch_name'] = test_epoch
+            # Xarray testing requires conversion to integer with ms default
+            self.testInst.data[test_epoch] = \
+                self.testInst[self.epoch_name].astype('int64') / 1.e6
+        else:
+            if 'epoch_name' in kwargs:
+                kwargs.pop('epoch_name')
+
```

Comment:
It looks like this test needs to be different for pandas and xarray.  What do you think about breaking it into two tests and skipping if the data format is the wrong type?  Or, if there base class is pandas and an xarray class inherits this, the inhertiting class can rewrite this test with the xarray version.

Reply:
The difference between these is that the pandas epoch is better integrated.  This could be updated as part of #991.  Needs a todo statement.